loader.js

```diff
@@ -86,47 +104,29 @@ async function processLayout(options, frontMatter, content, resourcePath, scans)
     '|'
   )})`
 
-  const extendedFm = await extendFrontMatter({
-    content,
-    frontMatter: {
-      ...frontMatter,
-      __resourcePath: resourcePath,
-      __scans: scans
-    },
-    phase: 'loader',
-    extendFm: pluginOpts.extendFrontMatter
-  })
 
   const matches = await new Promise((resolve, reject) => {
     glob(layoutMatcher, (err, matches) => err ? reject(err) : resolve(matches))
   })
 
   if (!matches.length) {
     throw new Error(
-      `File "${resourcePath}" specified "${frontMatter.layout}" as its layout, but no matching file was found at "${layoutMatcher}"`
+      `File "${resourcePath}" specified "${mergedFrontMatter.layout}" as its layout, but no matching file was found at "${layoutMatcher}"`
     )
   }
 
   const { onContent } = pluginOpts
   if (onContent && this._compiler.name === 'server') {
     onContent({
-      ...frontMatter,
-      ...extendedFm,
-      ...{ __resourcePath: resourcePath },
+      ...mergedFrontMatter,
       content
     })
```

Comment:
This section actually had a _slight_ bug where it didn't included `...{__scans: scans },`. It's fixed just by referencing `mergedFrontMatter` from above.